src/rust/cryptography-x509-verification/src/policy/mod.rs

```diff
@@ -329,6 +338,19 @@ impl<'a, B: CryptoOps> Policy<'a, B> {
     }
 
     fn permits_basic(&self, cert: &Certificate<'_>) -> Result<(), ValidationError> {
+        // CA/B 6.1.5: Key sizes
+        // NOTE: We don't currently enforce that RSA moduli are divisible by 8,
+        // since other implementations don't bother.
+        let cert_spki = &cert.tbs_cert.spki;
+        if cert_spki.algorithm.is_rsa() {
+            let rsa_key: Pksc1RsaPublicKey<'_> =
+                asn1::parse_single(cert_spki.subject_public_key.as_bytes())?;
+
+            if rsa_key.n.as_bytes().len() * 8 < self.minimum_rsa_modulus {
+                return Err(ValidationError::Other("RSA key is too weak".into()));
+            }
+        }
```

Comment:
This should probably be below the other checks.

Also is `permits_basic` the right place for this, or should it be a part of the signature algorithm checks. That is to say: Do we enforce a minimum key size for the EE key, even though it's never used in signature validation?

Reply:
Whoops, raced you on this; https://github.com/pyca/cryptography/pull/10302#discussion_r1472130273 has my rationale for this being in `permits_basic`. I'll move this below the other checks, though.

Reply:
I'll also check to see what other implementations do re: EE key sizes.

Reply:
Moved this to `permits_ca` since other implementations (besides OpenSSL, which decides to be special) don't enforce the key strength check on EEs.